### PR TITLE
Edits/intro analysis

### PR DESCRIPTION
**Pull request by smart-alek**
@ozan Getting back into this with a slew of edits for the intro/analysis.

The big thing is collapsing each paragraph into its own line. Prose shouldn't be subject to the same char limits as code, and Sublime has word wrapping anyway.

The rest of the edits are meant to cut down on verbosity with lessons I've been learning from writing docs at CircleCI. Most aggressive edits were made in the final chapter of Analysis (Performance of Python Types) in an attempt to break up some of those text blocks.

### CODE REVIEW

**Comment by smart-alek** [comment]
One more thing: I should probably apply the edits I made for the anagram example to "JavaScript" mode as well. Just a reminder for the books.

File: book/analysis/anagrams.py [hunk 1]
```diff
@@ -81,50 +61,27 @@ def anagram_sort_and_compare(s1, s2):
...
-ahead of time, so it will still generate $$n!$$ strings.
+There’s a problem with this approach, though: when generating all possible strings from `s1`, there are $$n$$ possible first characters, $$n-1$$ possible second characters, $$n-2$$ possible third characters, and so on. The total number of candidate strings is $$n*(n-1)*(n-2)*...*3*2*1$$, which is $$n!$$. Although some of the strings may be duplicates, the program won’t know that and will still generate $$n!$$ strings.
 
-It turns out that $$n!$$ grows even faster than $$2^{n}$$ as *n* gets large.
-If `s1` were 20 characters long, there would be $$20!$$ or
-2,432,902,008,176,640,000 possible candidate strings. If we processed one
-candidate every second, it would take 77,146,816,596 years to process the
-entire list. This is probably not going to be a good solution.
+It turns out that $$n!$$ grows even faster than $$2^{n}$$ as *n* gets large. If `s1` were 20 characters long, there would be $$20!$$ or
+2,432,902,008,176,640,000 possible candidate strings. If we processed one candidate every second, it would take 77,146,816,596 years to process the entire list. This is probably not going to be a good solution.
```
Comment (ozan) [inline book/analysis/anagrams.py:74]: @smart-alek how did you miss this linebreak? Tsk.

File: book/analysis/anagrams.py [hunk 2]
```diff
@@ -81,50 +61,27 @@ def anagram_sort_and_compare(s1, s2):
...
-a's, the same number of b's, the same number of c's, and so on. First, we
-generate character counts for each string. If these counts match, the two
-strings are anagrams.
+Our final solution uses the fact that any two anagrams have the same number of a’s, the same number of b’s, the same number of c’s, and so on. First, we generate character counts for each string. If these counts match, the two strings are anagrams.
 
-Since there are 26 possible characters, we can use a list of 26 counters for
-each string. Each time we see a particular character, we will increment the
-counter at that position. In the end, if the two lists are identical, the
-strings must be anagrams.
+Since there are 26 possible characters, we can use a list of 26 counters for each string. Each time we see a particular character, we’ll increment the counter at that character’s position. If the two lists are identical t the end, the strings must be anagrams.
```
Comment (ozan) [inline book/analysis/anagrams.py]: @smart-alek "identical t the end"

**Comment by ozan** [comment]
@smart-alek awesome, thanks for this! Reviewed and all looks good, just one typo above^. Will quickly fix myself and merge.

**Pull request merged**